tests/security/attributes/Writer.cpp

```diff
@@ -58,13 +58,18 @@ Writer::svc()
     ws->attach_condition(condition);
 
     DDS::Duration_t timeout =
-      { DDS::DURATION_INFINITE_SEC, DDS::DURATION_INFINITE_NSEC };
+      { 5, DDS::DURATION_INFINITE_NSEC };
```

Comment:
Rather than some arbitrary time, let's use a GuardCondition here and trigger it in Writer::end(). See performance-tests/bench/test_controller/ScenarioManager.cpp for an example of how to combine GuardCondition with other conditions on a WaitSet.